Aikido.Zen.DotNetFramework/HttpModules/ContextModule.cs

```diff
@@ -127,12 +127,23 @@ private static string GetClientIp(HttpContext httpContext)
                 : httpContext.Request.ServerVariables["REMOTE_ADDR"];
         }
 
+        /// <summary>
+        /// Gets a parameterized route from the HTTP context, matching against the route collection
+        /// and applying route parameter detection when needed.
+        /// </summary>
+        /// <param name="context">The HTTP context containing the request information</param>
+        /// <returns>A parameterized route string with a leading slash</returns>
         internal string GetParametrizedRoute(HttpContext context)
         {
             var routePattern = context.Request.Path;
             if (routePattern == null)
             {
-                return string.Empty;
+                return "/";
+            }
+
+            if (RouteHelper.PathIsSingleSegment(context.Request.Path))
```

Comment:
add a comment explaining this edge case with the example of /{slug}. Its not super obvious why we do this for outsiders.